test/node/src/aggregate-function.test.ts

```diff
@@ -0,0 +1,301 @@
+import { FunctionModule } from '../../../dist/cjs/index.js'
+import {
+  BUILT_IN_DIALECTS,
+  destroyTest,
+  initTest,
+  TestContext,
+  testSql,
+} from './test-setup.js'
+
+for (const dialect of BUILT_IN_DIALECTS) {
+  describe(`${dialect}: aggregate functions`, () => {
+    let ctx: TestContext
+
+    before(async function () {
+      ctx = await initTest(this, dialect)
+    })
+
+    after(async () => {
+      await destroyTest(ctx)
+    })
+
+    for (const func of ['avg', 'count', 'max', 'min', 'sum']) {
+      describe(func, () => {
+        it(`should execute a query with ${func}(...) in select clause`, async () => {
+          const query = ctx.db
+            .selectFrom('person')
+            .select([
+              (ctx.db.fn[func as keyof FunctionModule<any, any>] as Function)(
```

Comment:
I don't think you need these casts if you iterate over

```ts
for (const func of ['avg', 'count', 'max', 'min', 'sum'] as const) {
```

Reply:
Good catch! 

This removes the need for `as keyof FunctionModule`.
But ts still throws an error due to the key space not having a uniform function signature. 😐

Reply:
Don't know how to fix that one off the top of my head